lib/ansible/modules/cloud/amazon/aws_direct_connect_gateway.py

```diff
@@ -0,0 +1,375 @@
+    description:
+        - name of the dxgw to be created or deleted
+    required: false
+  amazon_asn:
+    description:
+        - amazon side asn
+    required: true
+  direct_connect_gateway_id:
+    description:
+        - id of an existing direct connect gateway
+    required: false
+  virtual_gateway_id:
+    description:
+        - vpn gateway id of an existing virtual gateway
+    required: false
+'''
+
+EXAMPLES = '''
+- name: Create a new direct connect gateway attached to virtual private gateway
+  dxgw:
+    state: present
+    name: my-dx-gateway
+    amazon_asn: 7224
+    virtual_gateway_id: vpg-12345
+  register: created_dxgw
+
+- name: Create a new unattached dxgw
+  dxgw:
+    state: present
+    name: my-dx-gateway
+    amazon_asn: 7224
+  register: created_dxgw
+
+'''
+
+RETURN = '''
+result:
+  description:
+    - The attributes of the Direct Connect Gateway
+  type: complex
+  returned: I(state=present)
+  contains:
+    amazon_side_asn:
+      description: ASN on the amazon side.
+    direct_connect_gateway_id:
+      description: The ID of the direct connect gateway.
+    direct_connect_gateway_name:
+      description: The name of the direct connect gateway.
+    direct_connect_gateway_state:
+      description: The state of the direct connect gateway.
+    owner_account:
+      description: The AWS account ID of the owner of the direct connect gateway.
+'''
+
+import time
+import traceback
+
+try:
+    import botocore
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (camel_dict_to_snake_dict, ec2_argument_spec, HAS_BOTO3,
+                                      get_aws_connection_info, boto3_conn)
+from ansible.module_utils.aws.direct_connect import DirectConnectError
+
+from ansible.module_utils._text import to_native
+
+
+def dx_gateway_info(client, gateway_id):
+    resp = client.describe_direct_connect_gateways(
+        directConnectGatewayId=gateway_id)
+    if resp is not []:
+        return resp['directConnectGateways'][0]
+    else:
+        None
+
+
+def wait_for_status(client, module, gateway_id, virtual_gateway_id, status):
+    polling_increment_secs = 15
+    max_retries = 3
+    status_achieved = False
+
+    for x in range(0, max_retries):
+        try:
+            response = check_dxgw_association(
+                client,
+                module,
+                gateway_id=gateway_id,
+                virtual_gateway_id=virtual_gateway_id)
+            if len(response['directConnectGatewayAssociations']) > 0:
+                if response['directConnectGatewayAssociations'][0]['associationState'] == status:
+                    status_achieved = True
+                    break
+                else:
+                    time.sleep(polling_increment_secs)
+            else:
+                status_achieved = True
+                break
+        except botocore.exceptions.ClientError as e:
```

Comment:
This (and following exception handling) should be `except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:`
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#boto3-1